File: frameworks/java/spring-boot/UnitTests/src/test/java/unittests/TemplateTests.java
package unittests;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.net.http.HttpResponse;
import java.nio.charset.StandardCharsets;

import org.junit.jupiter.api.Tag;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class TemplateTests extends SpringApp {

    // rb:test template.small,template.medium
    @ParameterizedTest
    @Tag("template.small")
    @Tag("template.medium")
    @ValueSource(strings = {"small", "medium"})
    void thymeleafRendersThePayloadAsTheCorpusPage(String size) throws Exception {
        HttpResponse<byte[]> response = get("/template/" + size);

        assertTrue(Answer.header(response, "content-type").startsWith("text/html"));
        assertEquals(Expected.normal(Expected.page("items." + size + ".json")), Expected.normal(new String(response.body(), StandardCharsets.UTF_8)));
    }
}

File: frameworks/dotnet/fastendpoints/UnitTests/TemplateTests.cs
namespace UnitTests;

public sealed class TemplateTests(App app) : TestBase<App>
{
    // rb:test template.small,template.medium
    [Theory]
    [Trait("corpus", "template.small")]
    [Trait("corpus", "template.medium")]
    [InlineData("small")]
    [InlineData("medium")]
    public async Task The_component_renders_the_payload_as_the_corpus_page(string size)
    {
        using HttpResponseMessage response = await app.Client.GetAsync($"/template/{size}", Cancellation);

        Assert.Equal("text/html", response.Content.Headers.ContentType?.MediaType);
        Assert.Equal(Expected.Normal(Expected.Page($"items.{size}.json")), Expected.Normal(await response.Content.ReadAsStringAsync(Cancellation)));
    }
}
